firmware/controllers/can/can.h

```diff
@@ -16,6 +16,25 @@
 
 #define CAN_TIMEOUT MS2NT(100)
 
+//can tx periodic task cycle time in frequency, 200hz -> 5ms period
+#define CAN_CYCLE_FREQ		(200)
+//can tx periodic task cycle time in ms
+#define CAN_CYCLE_PERIOD    (CH_CFG_ST_FREQUENCY / CAN_CYCLE_FREQ)
```

Comment:
is this actually in MS? `CH_CFG_ST_FREQUENCY` is just equal to the CPU clock.  I think you want `(1000.0f / CAN_CYCLE_FREQ)` to get ms.

Reply:
Freq is needed in the constructor, and the period constructor would actually reach out to a macro where CH_CFG_ST_FREQUENCY is divided by the parameter to get the needed ms. in the setPeriod method the param is ms, But yes, the freq needs to be float, i reused the macro no avoid constants.

Reply:
do we need `CAN_CYCLE_PERIOD` at all?